SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestRepNotify/SpatialTestRepNotify.cpp

```diff
@@ -24,47 +30,106 @@
+		RegisterAutoDestroyActor(TestActor);
+
+		TestActor->TestSubobject->OnChangedRepNotifyInt = 400;
+		TestActor->OnChangedRepNotifyInt1 = 350;
+
+		FinishStep();
+	});
+
+	AddStep(
+		TEXT("SpatialTestRepNotifyClientCheckOrderingWasCorrect"), FWorkerDefinition::AllClients, nullptr, nullptr,
+		[this](float DeltaTime) {
+			if (!RequireEqual_Bool(IsValid(TestActor), true, TEXT("TestActor should be valid.")))
+			{
+				return;
+			}
+			if (!RequireEqual_Bool(IsValid(TestActor->TestSubobject), true, TEXT("TestActor's subobject should be valid.")))
+			{
+				return;
+			}
+
+			RequireEqual_Int(TestActor->TestSubobject->OnChangedRepNotifyInt, 400,
+							 TEXT("The subobject's OnChangedRepNotifyInt property should have been updated to 400."));
+			RequireEqual_Int(TestActor->OnChangedRepNotifyInt1, 350,
+							 TEXT("The actors OnChangedRepNotifyInt1 property should have been updated to 350."));
+
+			RequireEqual_Bool(
+				TestActor->TestSubobject->bParentPropertyWasExpectedProperty, true,
+				TEXT("The OnChangedRepNotifyInt1 on parent actor ASpatialTestRepNotifyActor should have been set to 350 before the "
+					 "subobject's RepNotify was called."));
+			RequireEqual_Bool(TestActor->bSubobjectIntPropertyWasExpectedProperty, true,
+							  TEXT("The OnChangedRepNotifyInt1 on subobject USpatialTestRepNotifySubobject should have been set to 400 "
+								   "before the actor's RepNotify was called."));
+			FinishStep();
+		},
+		5.0f);
 
+	// Further core Rep Notify functionality testing
+	// The Server modifies the replicated variables.
+	AddStep(TEXT("SpatialTestRepNotifyServerChangeReplicatedVariables2"), FWorkerDefinition::Server(1), nullptr, [this]() {
+		TestActor->OnChangedRepNotifyInt1 = 1;
+		TestActor->AlwaysRepNotifyInt1 = 2;
+		TestActor->OnChangedRepNotifyInt2 = 3;
+		TestActor->AlwaysRepNotifyInt2 = 4;
+		TestActor->TestArray.Empty();
+		TestActor->TestArray.Add(1);
+		TestActor->TestArray.Add(2);
+		TestActor->TestSubobject->OnChangedRepNotifyInt = 2;
 		FinishStep();
 	});
 
 	// All clients check they received the correct values for the replicated variables.
 	AddStep(
 		TEXT("SpatialTestRepNotifyAllClientsCheckReplicatedVariables"), FWorkerDefinition::AllClients, nullptr, nullptr,
 		[this](float DeltaTime) {
-			if (bOnRepOnChangedRepNotifyInt1Called && bOnRepAlwaysRepNotifyInt1Called && OnChangedRepNotifyInt1 == 1
-				&& AlwaysRepNotifyInt1 == 2 && OnChangedRepNotifyInt2 == 3 && AlwaysRepNotifyInt2 == 4 && TestArray.Num() == 2
-				&& TestArray[0] == 1 && TestArray[1] == 2)
+			RequireEqual_Bool(TestActor->bOnRepOnChangedRepNotifyInt1Called, true,
+							  TEXT("Test actor's OnChangedInt1 Rep Notify should have been called"));
+			RequireEqual_Bool(TestActor->bOnRepAlwaysRepNotifyInt1Called, true,
+							  TEXT("Test actor's AlwaysInt1 Rep Notify should have been called"));
+
+			RequireEqual_Int(TestActor->OnChangedRepNotifyInt1, 1, TEXT("Test actor's OnChangedRepNotifyInt1 property should be 1"));
```

Comment:
I think just saying "Test actor's SomeProperty should have been updated" would be enough, as I think the passing and failing message provides the expected and actual value already.

Reply:
oki, changed these wordings, you are right - the less to change when it needs changing, the better